src/Components/Aspire.OpenAI/README.md

```diff
@@ -0,0 +1,123 @@
+
+```csharp
+private readonly OpenAIClient _client;
+
+public ChatController(OpenAIClient client)
+{
+    _client = client;
+}
+```
+
+To learn how to use the `OpenAIClient` class refer to [Using the OpenAIClient class](https://github.com/openai/openai-dotnet?tab=readme-ov-file#using-the-openaiclient-class).
+
+## Configuration
+
+The .NET Aspire OpenAI library provides multiple options to configure the OpenAI service based on the requirements and conventions of your project. Note that either an `Endpoint` or a `ConnectionString` is required to be supplied.
+
+### Use a connection string
+
+A connection can be constructed from an __Endpoint__ and a __Key__ value with the format `Endpoint={endpoint};Key={key};`. If the `Endpoint` value is omitted the default OpenAI endpoint will be used. You can provide the name of the connection string when calling `builder.AddOpenAIClient()`:
+
+```csharp
+builder.AddOpenAIClient("openaiConnectionName");
+```
+
+#### Connection string
+
+Alternatively, a custom connection string can be used.
+
+```json
+{
+  "ConnectionStrings": {
+    "openaiConnectionName": "Endpoint=https://{openai_rest_api_url}Key={account_key};"
+  }
+}
+```
+
+To obtain an API key on openai.com, first [create a new OpenAI account](https://platform.openai.com/signup) or [log in](https://platform.openai.com/login). 
+Next, navigate to the [API key page](https://platform.openai.com/account/api-keys) and select "Create new secret key", optionally naming the key.
+Make sure to save your API key somewhere safe and do not share it with anyone.
+
+### Use configuration providers
+
+The .NET Aspire OpenAI library supports [Microsoft.Extensions.Configuration](https://learn.microsoft.com/dotnet/api/microsoft.extensions.configuration). It loads the `OpenAISettings` and `OpenAIClientOptions` from configuration by using the `Aspire:OpenAI` key. Example `appsettings.json` that configures some of the options:
+
+```json
+{
+  "Aspire": {
+    "OpenAI": {
+      "DisableTracing": false,
+    }
+  }
+}
+```
+
+### Use inline delegates
+
+You can also pass the `Action<OpenAISettings> configureSettings` delegate to set up some or all the options inline, for example to disable tracing from code:
+
+```csharp
+builder.AddOpenAIClient("openaiConnectionName", settings => settings.DisableTracing = true);
+```
+
+You can also setup the `OpenAIClientOptions` using the optional `Action<OpenAIClientOptions>? configureOptions` parameter of the `AddOpenAIClient` method. For example, to set a custom `NetworkTimeout` value for this client:
+
+```csharp
+builder.AddOpenAIClient("openaiConnectionName", configureOptions: options => options.NetworkTimeout = TimeSpan.FromSeconds(2));
+```
+
+## AppHost extensions
+
+There is no specific AppHost extension corresponding to the OpenAI integration. Instead a connection string resource can be registered:
```

Comment:
For my learning -- what is an AppHost extension and is this something we would want to have a story for in the .NET ecosystem to support better ease of use for OpenAI?  Mostly just curious on this piece.

Reply:
An AppHost extension is a package that helps provision or describe the server part of the integration. For instance with Azure.AI.OpenAI there is a server resource that can be registered, will AI models too, such that we can provision the resource when the app is deployed. With OpenAI there is no such thing we can deploy, so we only declare a connection string, like a logical representation of this resource. Ultimately we could have a component that provides Ollama services for example.

Reply:
Thanks!  My comment here doesn't block this PR in any way, but I'm interested in understanding why we have one for Azure.AI.OpenAI and not OpenAI?  If there are pieces needed uniquely for SCM-based clients I'd be interested to understand the gaps there and if pieces are needed from SCM to address those gaps?

Reply:
> why we have one for Azure.AI.OpenAI and not OpenAI?

Because you can provision an Azure OpenAI service (using bicep). That's what the "AppHost" extension will do.

You can't provision an OpenAI service. You need to sign up on their website.